koku/masu/database/presto_sql/reporting_ocp_lineitem_daily_summary.sql

```diff
@@ -0,0 +1,379 @@
+/*
+ * Process OCP Usage Data Processing SQL
+ * This SQL will utilize Presto for the raw line-item data aggregating
+ * and store the results into the koku database summary tables.
+ */
+
+-- Using the convention of a double-underscore prefix to denote a temp table.
+
+/*
+ * ====================================
+ *               COMMON
+ * ====================================
+ */
+
+-- node label line items by day presto sql
+-- still using a "temp" table here because there is no guarantee how big this might get
+DROP TABLE IF EXISTS hive.{{schema | sqlsafe}}.__ocp_node_label_line_item_daily_{{uuid | sqlsafe}};
+CREATE TABLE hive.{{schema | sqlsafe}}.__ocp_node_label_line_item_daily_{{uuid | sqlsafe}} AS (
+    SELECT {{cluster_id}} as "cluster_id",
+           date(nli.interval_start) as "usage_start",
+           max(nli.node) as "node",
+           nli.node_labels,
+           max(nli.source) as "source",
+           max(nli.year) as "year",
+           max(nli.month) as "month"
+      FROM hive.{{schema | sqlsafe}}.openshift_node_labels_line_items as "nli"
+     WHERE nli.source = {{source}}
+       AND nli.year = {{year}}
+       AND nli.month in {{months | inclause}}
+       AND nli.interval_start >= TIMESTAMP {{start_date}}
+       AND nli.interval_start < date_add('day', 1, TIMESTAMP {{end_date}})
```

Comment:
We have a mix and match on the WHERE clauses in these files. My vote is to standardize on how it is in the rest of our files. Doing this does incur the cost of running the date() function more but also protects us if we ever don't pass in a date string. If it's a full TIMESTAMP we run the risk of not processing all the data in a day.

Reply:
The intent here was to try to lessen the work of casting data as it is read by making it 
```
START_TIMESTAMP = starting date/datetime
END_TIMESTAMP = ending date/datetime + 1 second

DATA_TIMESTAMP >= START_TIMESTAMP AND DATA_TIMESTAMP < END_TIMESTAMP
```
But if this is too confusing, then I'll change it back.

Reply:
It's not confusing, I was calling out that we've got mix-and-matched WHERE clauses, even in this file we're doing the same thing two different ways. 

My question with the method is what happens if we accidentally change our code to not pass in a date string, but a timestamp that has a non zero time component? We'll cleave some of the day's data.

Reply:
OK, had to look back at my code a sec.
So the call from Python is sending datetime.date values.
Since I'm using my mogrify func for parameter injection, it is being converted to a string **without** a cast (`date -> 'date'` instead of `date -> 'date'::date`
I'm only manipulating the parameters.
So in the start_date in presto becomes `TIMESTAMP '<date-str>'` which should evaluate to timestamp with zero time. And end_date in presto becomes `date_add('day', 1, TIMESTAMP '<date-str>')` 

Again this is to try to make the processing faster because the comparison parameters are now the same type as the inputs.

Reply:
Alright, a compromise, we'll need a guarantee in code that we can never pass in a timestamp with time data. https://issues.redhat.com/browse/COST-732

Reply:
It took completely reviewing my own code to realize the completely inconsistent way I was handling the date params. I've definitely made it consistent within and across files.

That stuff starts to all blend in together after awhile.

Reply:
This seemed like a good idea in that, as I recall, we were always passing dates and doing the timestamp manipulation of the parameters instead of _always_ manipulating stored data should be more efficient.

But, yeah. those inconsistencies needed to be addressed.